contracts/types/YieldSplitter.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import {IERC20} from "../interfaces/IERC20.sol";
+import {IgOHM} from "../interfaces/IgOHM.sol";
+import {SafeERC20} from "../libraries/SafeERC20.sol";
+
+/**
+    @title YieldSplitter
+    @notice Abstract contract that allows users to create deposits for their gOHM and have
+            their yield claimable by the specified recipient party. This contract's functions
+            are designed to be as generic as possible. This contract's responsibility is
+            the accounting of the yield splitting. All other logic such as error handling,
+            emergency controls, sending and recieving gOHM is up to the implementation of
+            this abstract contract to handle.
+ */
+abstract contract YieldSplitter {
+    using SafeERC20 for IERC20;
+
+    address public immutable gOHM;
+
+    struct DepositInfo {
+        uint256 id;
+        address depositor;
+        uint256 principalAmount; // Total amount of sOhm deposited as principal, 9 decimals.
+        uint256 agnosticAmount; // Total amount deposited priced in gOhm. 18 decimals.
+    }
+
+    uint256 public idCount;
+    mapping(uint256 => DepositInfo) public depositInfo; // depositId -> DepositInfo
+    mapping(address => uint256[]) public depositorIds; // address -> Array of the deposit id's deposited by user
+
+    /**
+        @notice Constructor
+        @param gOHM_ Address of gOHM.
+    */
+    constructor(address gOHM_) {
+        gOHM = gOHM_;
```

Comment:
I would remove this and just have the _toAgnostic and _fromAgnostic in this file using IsOHM's index. Maybe even don't use IsOHM directly and just define an interface at the top of this file:
```
interface IsOHM {
    function index() external view returns (uint256 index);
}```